src/tests/authenticator_test.ts

```diff
@@ -54,67 +55,174 @@ describe("Authenticator tests", () => {
+      const joseModule = require("jose");
+      const spy = jest
+        .spyOn(joseModule, "importPKCS8")
+        .mockImplementation(async () => { throw new Error("Import error"); });
       await expect(instance.buildJWT("https://example.com")).rejects.toThrow(
-        InvalidAPIKeyFormatError,
+        InvalidAPIKeyFormatError
       );
       await expect(instance.buildJWT("https://example.com")).rejects.toThrow(
-        "Could not parse the private key",
+        "Could not convert the EC private key to PKCS8 format"
       );
+      spy.mockRestore();
     });
 
-    test("should throw error if key type is not EC", async () => {
-      const mockKey = { kty: "RSA" };
-      jest.spyOn(JWK, "asKey").mockResolvedValue(mockKey as any);
-
+    test("should throw error if JWT signing fails", async () => {
+      const joseModule = require("jose");
+      jest.spyOn(joseModule, "importPKCS8").mockResolvedValue({} as any);
+      const spy = jest
+        .spyOn(SignJWT.prototype, "sign")
+        .mockRejectedValue(new Error("Signing error"));
+      await expect(instance.buildJWT("https://example.com")).rejects.toThrow(
+        InvalidAPIKeyFormatError
+      );
       await expect(instance.buildJWT("https://example.com")).rejects.toThrow(
-        InvalidAPIKeyFormatError,
+        "Could not convert the EC private key to PKCS8 format"
       );
+      spy.mockRestore();
     });
+  });
+});
 
-    test("should throw error if JWT signing fails", async () => {
-      const mockKey = { kty: "EC" };
-      jest.spyOn(JWK, "asKey").mockResolvedValue(mockKey as any);
-      const mockSign = {
-        update: jest.fn().mockReturnThis(),
-        final: jest.fn().mockRejectedValue(new Error("Signing error")),
-      };
-      jest.spyOn(JWS, "createSign").mockReturnValue(mockSign as any);
+describe("Authenticator tests for Edwards key", () => {
+  const filePath = "./config/test_ed25519_api_key.json";
+  const keys = require(filePath);
+  let authenticator: CoinbaseAuthenticator;
+  let source: string;
+  let sourceVersion: string | undefined;
+  let privateKey: string;
+  let apiKey: string;
 
-      await expect(instance.buildJWT("https://example.com")).rejects.toThrow(
-        InvalidAPIKeyFormatError,
-      );
+  beforeEach(() => {
+    // Use the Edwards key from config (a valid Base64-encoded 64-byte string)
+    privateKey = keys.privateKey;
+    apiKey = keys.name;
+    source = "mockSource";
+    sourceVersion = undefined;
+    jest.spyOn(console, "log").mockImplementation(() => {});
+    authenticator = new CoinbaseAuthenticator(apiKey, privateKey, source, sourceVersion);
+  });
+
+  it("should raise InvalidConfiguration error for invalid config", async () => {
+    const invalidConfig = {
+      method: "GET",
+      url: "", 
+      headers: {} as AxiosHeaders,
+    };
+    await expect(authenticator.authenticateRequest(invalidConfig)).rejects.toThrow();
+  });
+
+  describe("Edwards branch", () => {
```

Comment:
Curious what this describe block means? Not sure what "branch" is referring to?

Is this just `describe("witha valid configuration"...`?